From 12384828d7e030fa67bf1943aca4e14e95296e8c Mon Sep 17 00:00:00 2001
From: Sam Gwydir <sam.gwydir@joyent.com>
Date: Tue, 12 Dec 2017 21:21:08 +0000
Subject: [PATCH] OS-5927 postmortem DTrace frequently broken under vmware
 Reviewed by: Tim Kordas <tim.kordas@joyent.com> Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Approved by: Patrick Mooney
 <patrick.mooney@joyent.com>

---
 usr/src/lib/libdtrace/common/dt_consume.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/usr/src/lib/libdtrace/common/dt_consume.c b/usr/src/lib/libdtrace/common/dt_consume.c
index 7f8c673dbe..549abb2e75 100644
--- a/usr/src/lib/libdtrace/common/dt_consume.c
+++ b/usr/src/lib/libdtrace/common/dt_consume.c
@@ -24,7 +24,7 @@
  */
 
 /*
- * Copyright (c) 2013, Joyent, Inc. All rights reserved.
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
  * Copyright (c) 2012 by Delphix. All rights reserved.
  */
 
@@ -3013,9 +3013,6 @@ dtrace_consume(dtrace_hdl_t *dtp, FILE *fp,
 				break;
 
 			timestamp = dt_buf_oldest(buf, dtp);
-			assert(timestamp >= dtp->dt_last_timestamp);
-			dtp->dt_last_timestamp = timestamp;
-
 			if (timestamp == buf->dtbd_timestamp) {
 				/*
 				 * We've reached the end of the time covered
@@ -3029,6 +3026,8 @@ dtrace_consume(dtrace_hdl_t *dtp, FILE *fp,
 					break;
 				continue;
 			}
+			assert(timestamp >= dtp->dt_last_timestamp);
+			dtp->dt_last_timestamp = timestamp;
 
 			if ((rval = dt_consume_cpu(dtp, fp,
 			    buf->dtbd_cpu, buf, B_TRUE, pf, rf, arg)) != 0)
-- 
2.21.0

